### PR TITLE
feat: implement `container_pid_to_host_pid()`

### PR DESCRIPTION
**Pull request by kyujin-cho**
This PR implements `ai.backend.agent.utils.container_pid_to_host_pid()` function, which has been left as TODO.

### CODE REVIEW

File: src/ai/backend/agent/utils.py [hunk 1]
```diff
@@ -235,63 +288,26 @@ async def host_pid_to_container_pid(container_id: str, host_pid: HostPID) -> Con
...
+
+                container_pid = ContainerPID(PID(container_pids[process_idx]))
                 log.debug("host pid {} is mapped to container pid {}", host_pid, container_pid)
-                return ContainerPID(PID(int(container_pid)))
+                return container_pid
             except asyncio.CancelledError:
                 raise
-            except (IndexError, KeyError, aiodocker.exceptions.DockerError):
+            except (IndexError, KeyError, aiodocker.exceptions.DockerError) as e:
+                print(type(e), e)
```
Comment (achimnol) [inline src/ai/backend/agent/utils.py]: Remove a debug print.

File: src/ai/backend/agent/utils.py [hunk 2]
```diff
@@ -321,8 +337,46 @@ async def host_pid_to_container_pid(container_id: str, host_pid: HostPID) -> Con
...
+                host_procs = await get_host_process_table(docker, container_id)
+                host_pids = [x.pid for x in host_procs if x.command == cmd]
+
+                host_pid = HostPID(PID(host_pids[process_idx]))
+                log.debug("container pid {} is mapped to host pid {}", container_pid, host_pid)
+                return host_pid
+            except asyncio.CancelledError:
+                raise
+            except (IndexError, KeyError, aiodocker.exceptions.DockerError) as e:
+                print(type(e), e)
```
Comment (achimnol) [inline src/ai/backend/agent/utils.py]: Same here. :wink:

**Review by achimnol — CHANGES_REQUESTED** [review]

**Comment by sanxiyn** [comment]
This PR needs an update for cgroup v2 support(#865).

**Review by achimnol — APPROVED** [review]
It's a good refactoring.

Though, `get_conatiner_process_table()` could be reimplemented using `/proc/{pid}/status`'s `NSPid` field (ref: https://man7.org/linux/man-pages/man5/proc.5.html) following `get_host_process_table()`, as long as we're on the Linux kernel 4.1 or later.
I think we could rewrite `get_container_process_table()` as a follow-up PR.

**Pull request merged**